api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerInstanceTypePriceTypeAndCloudRegionTest.java

```diff
@@ -0,0 +1,707 @@
+
+    @Mock
+    private ToolManager toolManager;
+
+    @Mock
+    private AuthManager securityManager;
+
+    @Mock
+    private RunPermissionManager permissionManager;
+
+    @Mock
+    private DataStorageManager dataStorageManager;
+
+    @Mock
+    private PipelineConfigurationManager pipelineConfigurationManager;
+
+    @Mock
+    private MessageHelper messageHelper;
+
+    @Mock
+    private InstanceOfferManager instanceOfferManager;
+
+    @Mock
+    private PipelineLauncher pipelineLauncher;
+
+    @Mock
+    private CloudRegionManager cloudRegionManager;
+
+    @Mock
+    private ToolVersionManager toolVersionManager;
+
+    @Mock
+    private CheckPermissionHelper permissionHelper;
+
+    @Mock
+    private PipelineRunDao pipelineRunDao;
+
+    @Mock
+    private PreferenceManager preferenceManager;
+
+    private static final String TEST_IMAGE = "testImage";
+    private Tool tool;
+    private PipelineConfiguration configuration;
+    private PipelineConfiguration configurationWithoutRegion;
+    private InstancePrice price;
+    private AwsRegion defaultAwsRegion;
+    private AwsRegion nonAllowedAwsRegion;
+    private AwsRegion notDefaultAwsRegion;
+    private PipelineRun parentRun;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        tool = getTool(TEST_IMAGE, DEFAULT_COMMAND);
+        defaultAwsRegion = getDefaultAwsRegion(ID);
+        nonAllowedAwsRegion = getNonDefaultAwsRegion(ID_2);
+        notDefaultAwsRegion = getNonDefaultAwsRegion(ID_3);
+        configuration = getPipelineConfiguration(TEST_IMAGE, INSTANCE_DISK, true, defaultAwsRegion.getId());
+        configurationWithoutRegion = getPipelineConfiguration(TEST_IMAGE, INSTANCE_DISK);
+        price = new InstancePrice(configuration.getInstanceType(), parseInt(configuration.getInstanceDisk()),
+                PRICE_PER_HOUR, COMPUTE_PRICE_PER_HOUR, DISK_PRICE_PER_HOUR);
+        parentRun = PipelineCreatorUtils.getPipelineRunWithInstance(PARENT_RUN_ID, TEST_USER, ID_3);
+
+        mockCloudRegionManager();
+        mockInstanceOfferManager();
+        mockPreferenceManager();
+        mockPipelineConfigurationManager();
+        mockToolManager();
+
+        doReturn(parentRun).when(pipelineRunDao).loadPipelineRun(eq(PARENT_RUN_ID));
+        doReturn(false).when(permissionManager).isRunSshAllowed(eq(parentRun));
+        doReturn(DEFAULT_COMMAND).when(pipelineLauncher).launch(
+                any(PipelineRun.class), any(), any(), anyString(), anyString());
+        doReturn(ToolVersion.builder().size(1L).build())
+                .when(toolVersionManager).loadToolVersion(anyLong(), anyString());
+        doReturn(true).when(permissionHelper).isAllowed(any(), any());
+        doReturn(TEST_USER).when(securityManager).getAuthorizedUser();
+    }
+
+    @Test
+    public void shouldLaunchPipelineValidatesToolInstanceTypeAndPriceType() {
```

Comment:
The idea to have long test case names is to be descriptive. For this test and all the others I would propose using names which are almost english sentences like `launchToolShouldValidateToolInstanceTypeAndPriceType`.

Reply:
Done!